src/mem/epoch/markable_atomic.rs

```diff
@@ -48,32 +53,53 @@ fn opt_owned_into_usize<T>(ptr: Option<Owned<T>>, mark: usize) -> usize {
     raw as usize | mark
 }
 
-impl<T> MarkedAtomic<T> {
-    /// Create a new, null atomic pointer.
+impl<T> MarkableAtomic<T> {
+    /// Create a new, null markable atomic pointer with a marker value of 0.
     #[cfg(feature = "nightly")]
-    pub const fn null() -> MarkedAtomic<T> {
-        MarkedAtomic {
+    pub const fn zero() -> MarkableAtomic<T> {
+        MarkableAtomic {
             ptr: atomic::AtomicUsize::new(0),
-            _marker: PhantomData
+            _marker: PhantomData,
         }
     }
 
+    /// Create a new, null markable atomic pointer with a marker value of 0.
     #[cfg(not(feature = "nightly"))]
-    pub fn null() -> MarkedAtomic<T> {
-        MarkedAtomic {
+    pub fn zero() -> MarkableAtomic<T> {
+        MarkableAtomic {
             ptr: atomic::AtomicUsize::new(0),
-            _marker: PhantomData
+            _marker: PhantomData,
+        }
+    }
+
+    /// Create a new, null markable atomic pointer with the given marker value.
+    pub fn null(mark: usize) -> MarkableAtomic<T> {
+        guard_mark::<T>(mark);
+        MarkableAtomic {
+            ptr: atomic::AtomicUsize::new(mark),
+            _marker: PhantomData,
         }
     }
 
-    /// Create a new atomic pointer
-    pub fn new(data: T, mark: usize) -> MarkedAtomic<T> {
+    /// Create a new markable atomic pointer.
+    pub fn new(data: T, mark: usize) -> MarkableAtomic<T> {
         guard_mark::<T>(mark);
-        MarkedAtomic {
+        MarkableAtomic {
             ptr: atomic::AtomicUsize::new(
                 Box::into_raw(Box::new(data)) as usize | mark
             ),
-            _marker: PhantomData
+            _marker: PhantomData,
+        }
+    }
+
+    /// Unsafely create a new markable atomic pointer from the given raw pointer.
```

Comment:
Additions to documentation:

- Mention that this panics if the mark is too large.
- Explain why this is unsafe (like `Atomic`, a `MarkedAtomic` must either be null or be dereferenceable).